code/modules/interface/multiContext/context_actions.dm

```diff
@@ -782,8 +782,136 @@
 
 		execute(var/atom/target, var/mob/user)
 			var/obj/item/c = target
+			if(c.loc == user)
+				user.u_equip(c)
 			user.put_in_hand_or_drop(c)
 
+/datum/contextAction/card
+	icon = 'icons/ui/context16x16.dmi'
+	name = "Card action"
+	desc = "You shouldn't be reading this, bug."
+	icon_state = "wrench"
+	
+	checkRequirements(var/atom/target, var/mob/user)
+		return 1
+
+	solitaire
+		name = "solitaire stack"
+		desc = "stacks cards with a slight offset."
+		icon_state = "solitaire"
+
+		execute(var/atom/target, var/mob/user)
+			var/obj/item/playing_card/c = target
```

Comment:
please use something other than `c`, i know these are short, but pls

applies to whole file